src/client/entity-editor/work-section/work-section-merge.tsx

```diff
@@ -48,7 +48,7 @@ type StateProps = {
 };
 
 type DispatchProps = {
-	onTypeChange: (?{value: number}) => mixed
+	onTypeChange: (value: number | null) => unknown
```

Comment:
In other file, I believe you turned `?{value: number}` into `obj: {value: number} | null` instead, which I would even turn into `obj?: {value: number}`

Same for line 118 I suppose

Reply:
A lot of these event handlers had incorrect typing with Flow. Most of the event handlers typed to receive an object parameter actually took a single value - you can tell from how it was being used in the dispatch function:

	return {
		onTypeChange: (value: ?{value: number}) =>
			dispatch(updateType(value))
	};

And then in the actions:

    export function updateType(newTypeId: ?number): Action {

So there was a conflict there, in that the updateType action creator is expecting a single numeric ID rather than an object. 

To back this up, the event is coming from `MergeField`, which internally uses a `Select` component from `react-select`. The Select component returns an object with `value` and `label` keys, which `MergeField` internally filters to extract the `value` and pass this directly to the `MergeField.onChange` handler. So it's right that the `onChange` handler for MergeField takes a single value, while the onChange handler in the original `WorkSection` component takes an object.